src/command/process_update_task.rs

```diff
@@ -162,46 +298,255 @@ impl UpdateProcessor {
             .await?;
 
         self.repo
-            .modify_state(&self.chat.id, self.chat.user_id, ClientState::Number)
+            .modify_state(
+                &self.chat.id,
+                self.chat.user_id,
+                ClientState::FindCityNumber,
+            )
+            .await?;
+
+        Ok(())
+    }
+
+    async fn process_set_city(&self) -> Result<(), BotError> {
+        self.find_city().await?;
+
+        self.repo
+            .modify_selected(&self.chat.id, self.chat.user_id, self.text.clone())
+            .await?;
+
+        self.repo
+            .modify_state(&self.chat.id, self.chat.user_id, ClientState::SetCityNumber)
             .await?;
 
         Ok(())
     }
 
-    async fn process_number(&self) -> Result<(), BotError> {
+    async fn process_find_city_number(&self) -> Result<(), BotError> {
         match self.text.parse::<usize>() {
             Ok(number) => {
+                let city = self
+                    .repo
+                    .get_city_row(&self.chat.selected.clone().unwrap(), number)
+                    .await?;
+
                 self.return_to_initial().await?;
 
-                self.pattern_response(number).await
+                self.get_weather(city).await
             }
 
             Err(_) => self.not_number_message().await,
         }
     }
 
-    async fn pattern_response(&self, number: usize) -> Result<(), BotError> {
-        let city = self
+    async fn process_set_city_number(&self) -> Result<(), BotError> {
+        match self.text.parse::<usize>() {
+            Ok(number) => {
+                let city = self
+                    .repo
+                    .get_city_row(&self.chat.selected.clone().unwrap(), number)
```

Comment:
let's return error instead of unwrap

Reply:
I can not use `?` operator because `chat.selected` is an Option.